### PR TITLE
Bundle in solution dir

### LINKED ISSUE #5912: Bundling is broken in Release for apps not at the root of a project
We define a `BundleConfigFile` (to empty) and never use it

https://github.com/microsoft/react-native-windows/blob/c4025b80c454a06eb9d86958cab7487c4981aae9/vnext/PropertySheets/Bundle.props#L47

I am hitting an issue where running metro (in debug) picks up the metro.config.js by default as expected, but `npx react-native bundle` doesn't pick it up. It does only if I pass `--config metro.config.js`

### LINKED ISSUE #6687: Cannot build release e2eTest in Visual Studio
In VS, OutDir is a relative path for C# projects, e.g. `bin\x86\Release`
```
    <BundleSourceMapDir Condition="'$(BundleSourceMapDir)' == ''">$([MSBuild]::NormalizePath('$(OutDir)\sourcemaps\react'))</BundleSourceMapDir>
```

Therefore `BundleSourceMapDir` ends up wrongly pointing under the VS install dir

### PR DESCRIPTION
**Pull request by asklar**
Fixes #6687 
Fixes #5912 by removing the BundleConfigFile property which wasn't hooked up to anything.

When building release, the bundle props file specifies `OutDir` which is a relative paths (bin\x86\Release, for example). Therefore the full path will be under the current working directory. For `run-windows`, that's the app's folder, but when building in VS, it will be the VS install dir, or C:\windows\system32, or whatever the CWD was when VS was invoked, depending on how it was invoked.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6736)

### CODE REVIEW

File: vnext/PropertySheets/Bundle.props [hunk 1]
```diff
@@ -50,14 +50,17 @@
     <BundleEntryFile Condition="'$(BundleEntryFile)' == ''">index.js</BundleEntryFile>
 
     <!-- Optional config file for the bundle command (metro will use metro.config.js by default) -->
-    <BundleConfigFile Condition="'$(BundleConfigFile)' == ''"></BundleConfigFile>
+    <BundleConfigFile Condition="'$(BundleConfigFile)' == ''"/>
```
Comment (vmoroz) [inline vnext/PropertySheets/Bundle.props:53]: >BundleConfigFile [](start = 5, length = 16)

Why is it needed? It says: if BundleConfigFile was empty then make it empty. The net result is the same as we just remove this line.

File: vnext/PropertySheets/Bundle.props [hunk 2]
```diff
@@ -36,7 +36,7 @@
     <UseDevBundle Condition="'$(UseDevBundle)' == '' and '$(Configuration)' != 'Debug'">false</UseDevBundle>
 
     <!-- Extra arguments to pass to the bundler command -->
-    <BundlerExtraArgs Condition="'$(BundlerExtraArgs)' == '' AND '$(UseHermes)' != 'true'"></BundlerExtraArgs>
+    <BundlerExtraArgs Condition="'$(BundlerExtraArgs)' == '' AND '$(UseHermes)' != 'true'"/>
```
Comment (vmoroz) [inline vnext/PropertySheets/Bundle.props:39]: ><BundlerExtraArgs Condition="'$(BundlerExtraArgs)' == '' AND '$(UseHermes)' != 'true'"/> [](start = 4, length = 88)

This line doe snot nothing. Could we remove it?

File: vnext/PropertySheets/Bundle.props [hunk 3]
```diff
@@ -73,7 +76,7 @@
     <HermesCompilerCommand Condition="'$(HermesCompilerCommand)' == ''">$(HermesPackage)\tools\hermes.exe</HermesCompilerCommand>
 
     <!-- Flags passed to Hermes when compiling a bundle -->
-    <HermesCompilerFlags Condition="'$(HermesCompilerFlags)' == '' AND '$(Configuration)' == 'Debug'"></HermesCompilerFlags>
+    <HermesCompilerFlags Condition="'$(HermesCompilerFlags)' == '' AND '$(Configuration)' == 'Debug'"/>
```
Comment (vmoroz) [inline vnext/PropertySheets/Bundle.props:79]: >HermesCompilerFlags [](start = 5, length = 19)

It seems to be not useful too.

**Comment by vmoroz** [comment]
We had a behavior change in beachball that we run on `yarn change`. Now we must explicitly provide the branch against we do the change.
Please remove all the new change files and run `yarn change --branch upstream/maser`.

Note that beachball behaves correctly in 0.63-stable branch. The issue started in master and 0.64 branches.

**Review by vmoroz — CHANGES_REQUESTED** [review]
:clock1:

File: vnext/PropertySheets/Bundle.props (hunk 1 above)
Comment (asklar) [reply vnext/PropertySheets/Bundle.props:53]: This part of the change (changing it to self-closing tag) was made by VS. I think the original intent of the line was to express that this is a property that can be set.

**Comment by asklar** [comment]
> We had a behavior change in beachball that we run on `yarn change`. Now we must explicitly provide the branch against we do the change.
> Please remove all the new change files and run `yarn change --branch upstream/maser`.
> 
> Note that beachball behaves correctly in 0.63-stable branch. The issue started in master and 0.64 branches.
> 
> Though, I may not be correct here. Please let me know if the big number of change files is expected here and not due to the bug in beachball.

thanks for the heads up... was there an email about this that I missed? are we going to fix this to automatically detect the branch to compare against? seems cumbersome to require specifying what branch to use

File: vnext/PropertySheets/Bundle.props (hunk 1 above)
Comment (asklar) [reply vnext/PropertySheets/Bundle.props:53]: Having said that, nothing is actually consuming this flag (see #5912) so removing this flag

**Comment by vmoroz** [comment]
> > We had a behavior change in beachball that we run on `yarn change`. Now we must explicitly provide the branch against we do the change.
> > Please remove all the new change files and run `yarn change --branch upstream/maser`.
> > Note that beachball behaves correctly in 0.63-stable branch. The issue started in master and 0.64 branches.
> > Though, I may not be correct here. Please let me know if the big number of change files is expected here and not due to the bug in beachball.
> 
> thanks for the heads up... was there an email about this that I missed? are we going to fix this to automatically detect the branch to compare against? seems cumbersome to require specifying what branch to use

I have experienced this issue myself. It was discussed internally, but I do not think there are any plans to fix it. We can discuss it offline.

File: vnext/PropertySheets/Bundle.props [hunk 4]
```diff
@@ -49,15 +48,15 @@
     <BundleEntryFile Condition="'$(BundleEntryFile)' == '' and Exists('$(BundleCommandWorkingDir)\index.windows.js')">index.windows.js</BundleEntryFile>
     <BundleEntryFile Condition="'$(BundleEntryFile)' == ''">index.js</BundleEntryFile>
 
-    <!-- Optional config file for the bundle command (metro will use metro.config.js by default) -->
```
Comment (jonthysell) [inline vnext/PropertySheets/Bundle.props:52]: Is there a reason this is being removed?

File: vnext/PropertySheets/Bundle.props [hunk 5]
```diff
@@ -49,15 +48,15 @@
...
-    <BundleConfigFile Condition="'$(BundleConfigFile)' == ''"></BundleConfigFile>
-
+    <FullOutDir Condition="'$([System.IO.Path]::GetFullPath($(OutDir)))' == '$(OutDir)'">$(OutDir)</FullOutDir>
```
Comment (jonthysell) [inline vnext/PropertySheets/Bundle.props:51]: I'm not getting what the desired logic is. Are you checking if the path is relative or not?

**Review by jonthysell — APPROVED** [review]

File: vnext/PropertySheets/Bundle.props (hunk 4 above)
Comment (asklar) [reply vnext/PropertySheets/Bundle.props:52]: this was never hooked up to anything, nobody is actually using it.

File: vnext/PropertySheets/Bundle.props (hunk 5 above)
Comment (asklar) [reply vnext/PropertySheets/Bundle.props:51]: yes. For C# these seem to be in the form bin\x86\Release, whereas for vcxproj they seem to be full paths. However there is no requirement that this be the case and the doc for this variable says "if it is a relative path, it's relative to the project dir"

**Review by vmoroz — APPROVED** [review]
:shipit:

**Review by dannyvv — APPROVED** [review]

**Pull request merged**